draft-gpew-priv-ppm.md

```diff
@@ -958,8 +1089,91 @@ responds with HTTP status 202 Accepted. The response MAY include a Retry-After
+
+Then, for each aggregator endpoint `[aggregator]` in the parameters associated
+with `CollectReq.task_id` (see {{collect-flow}}) except its own, the leader sends
+a POST request to `[aggregator]/aggregate_share` with the following message:
+
+~~~
+struct {
+  TaskID task_id;
+  Interval batch_interval;
+  uint64 report_count;
+  opaque checksum[32];
+  opaque helper_state<0..2^16>;
+} AggregateShareReq;
+~~~
+
+* `task_id` is the task ID associated with the PPM parameters.
+* `batch_interval` is the batch interval of the request.
+* `report_count` is the number of reports included in the aggregation.
+* `checksum` is the checksum computed over the set of client reports. The
+  checksum is computed by taking the SHA256 hash of each nonce from the client
+  reports included in the aggregation, then combining the hash values with a
+  bitwise-XOR operation.
+* `helper_state` is the helper's state, which is carried across requests from
+  the leader.
+
+To handle the leader's request, the helper first ensures that the request meets
+the requirements for batch parameters following the procedure in
+{{batch-parameter-validation}}. If the batch parameters are invalid, then it
+MUST abort with error "batchInvalid".
+
+Next, it computes a checksum based on its view of the output shares included in
+the batch window, and checks that the `report_count` and `checksum` included in
+the request match its computed values. If not, then it MUST abort with error
+"batchMismatch".
+
+Next, it computes the aggregate share for the batch interval as described in
+{{agg-complete}}, obtaining an opaque byte string `agg_share` whose
+structure is specific to the VDAF in use. The helper encrypts the
+`agg_share` as follows:
+
+~~~
+enc, context = SetupBaseS(pk, AggregateShareReq.task_id ||
+                              "ppm-00 aggregate share" || 0x03 || 0x00)
+
+encrypted_agg_share = context.Seal(AggregateShareReq.batch_interval,
+                                   agg_share)
+~~~
+
+where `pk` is the HPKE public key encoded by the collector's HPKE key.
+
+Encryption prevents the leader from learning the actual result, as it only has
+its own aggregate share and cannot compute the helper's.
+
+The helper responds to the leader with HTTP status 200 OK and a body consisting
+of an `AggregateShareResp`:
+
+~~~
+struct {
+    HpkeCiphertext encrypted_aggregate_share;
+} AggregateShareResp;
+~~~
+
+`encrypted_aggregate_share.config_id` is set to the collector's HPKE config ID.
+`encrypted_aggregate_share.enc` is set to the encapsulated HPKE context `enc`
+computed above and `encrypted_aggregate_share.ciphertext` is the ciphertext
+`encrypted_agg_share` computed above.
+
+After receiving the helper's response, the leader uses the HpkeCiphertext to
+respond to a collect request (see {{collect-flow}}).
+
+The leader MAY make multiple aggregate-share requests for a given batch interval
+and aggregation parameter MUST get the same result each time.
```

Comment:
```suggestion
The leader MAY make multiple aggregate-share requests for a given batch interval
and aggregation parameter and MUST get the same result each time.
```
The intent here is that the helper should use the same set of reports to service multiple aggregate-share requests, right? Could that be stated explicitly.

Reply:
The intent here is that if the parameters of the aggregation request are the same then the output should be the same.